src/names.c

```diff
@@ -90,102 +95,95 @@ SEXP as_unique_names(SEXP names, bool quiet) {
   }
 
   R_len_t i = 0;
-  R_len_t n = Rf_length(names);
-  SEXP* ptr = STRING_PTR(names);
+  const R_len_t n = Rf_length(names);
+  SEXP* const names_ptr = STRING_PTR(names);
```

Comment:
```suggestion
  const SEXP* names_ptr = STRING_PTR_RO(names);
```

Reply:
First mention of `STRING_PTR_RO` in wch/r-source@6ac7b5670cd (Nov 2017), but now I see we have compat code. Neat!

This is a constant pointer to (modifiable) `SEXP`, not a modifiable pointer to const `SEXP`: https://stackoverflow.com/a/1143272/946850. If we chose to switch to const qualifiers everywhere, we would write:

```suggestion
  SEXP const * const names_ptr = STRING_PTR_RO(names);
```

Reply:
I don't want to mark the pointers as const in this codebase.

Reply:
What's important here is that the array can't change:

```c
const SEXP* data = STRING_PTR_RO(i);

data[3] = R_NilValue;

   slice.c:408:9: error: read-only variable is not assignable
     ip[3] = R_NilValue;
     ~~~~~ ^
   1 error generated.
```